### PR TITLE
ETCD-385: Prefer ipv6 on bootstrap in v6-primary dual stack

### PR DESCRIPTION
**Pull request by cybertron**
Previously we only supported v4-primary dual stack, so it was correct to always use ipv4 unless we were single stack v6. Now we need to add support for v6-primary dual stack so we need to be more nuanced about ip selection.

This change looks at the first CIDR to determine whether the cluster should be v4 or v6 primary. The other logic stays essentially the same since it was already set up to look for ipv6 addresses when required.

### CODE REVIEW

File: pkg/cmd/render/render.go [hunk 1]
```diff
@@ -621,18 +621,16 @@ func writeManifests(outputDir, templateDir string, templateData interface{}) err
...
 	}
-	for _, cidr := range serviceCIDRs {
-		ip, _, err := net.ParseCIDR(cidr)
-		if err != nil {
-			return false, err
-		}
-		if ip.To4() != nil {
-			return false, nil
-		}
+	ip, _, err := net.ParseCIDR(serviceCIDRs[0])
```
Comment (tjungblu) [inline pkg/cmd/render/render.go:628]: can you please leave a comment here why we do that? 

maybe with a quick example of a serviceCIDR listing and it's desired behavior in terms of bootstrap IPs

File: pkg/cmd/render/render.go [hunk 2]
```diff
@@ -621,18 +621,16 @@ func writeManifests(outputDir, templateDir string, templateData interface{}) err
 	return nil
 }
 
-func isSingleStackIPv6(serviceCIDRs []string) (bool, error) {
+func preferIPv6(serviceCIDRs []string) (bool, error) {
```
Comment (tjungblu) [inline pkg/cmd/render/render.go:624]: would you mind adding a quick unit test for this func specifically? just testing a couple of input/output pairs

Comment (cybertron) [reply pkg/cmd/render/render.go:624]: Yep, will do.

File: pkg/cmd/render/render.go (hunk 1 above)
Comment (cybertron) [reply pkg/cmd/render/render.go:628]: Sure.

**Review by cybertron — COMMENTED** [review]
Will update this shortly to address the inline comments.

File: pkg/cmd/render/render.go (hunk 1 above)
Comment (tjungblu) [reply pkg/cmd/render/render.go:645]: perfect, thanks!

File: pkg/cmd/render/render.go (hunk 2 above)
Comment (tjungblu) [reply pkg/cmd/render/render.go:624]: awesome, thanks again.

**Comment by tjungblu** [comment]
@cybertron I think there's just a formatting issue left:

> verify-gofmt failed - please run `make update-gofmt`


otherwise I'm lgtm :)

**Comment by cybertron** [comment]
Oops, fixed!

**Comment by Elbehery** [comment]
@cybertron do we need backports ?

**Comment by cybertron** [comment]
> @cybertron do we need backports ?

No, this is for a new feature in 4.13.

**Pull request merged**